### ISSUE TITLE
I am able to run callbacks using model.fit. The corresponding code is                '' model.fit(X_train, Y_train,               batch_size=batch_size,               nb_epoch=nb_epoch,               shuffle=True,               verbose=1,              validation_data=(X_valid, Y_valid),callbacks=[save_model,          early_stopping,csv_logger,qwk]'' . How to use callbacks with model.fit_generator? It is showing error 'fit_generator() takes at least 3 arguments (6 given).' The corresonding code is                                            '' model.fit_generator(datagen.flow(X_train, Y_train, batch_size=batch_size,shuffle=True,save_to_dir=aug_path, save_prefix='aug'),                     nb_epoch=nb_epoch,verbose=1,                     validation_data=(X_valid, Y_valid),                     callbacks=[save_model,          early_stopping,csv_logger,qwk])''

### ISSUE THREAD

**Issue by Niladri365** [keras-team/keras#7699]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by netrapathak** [comment]
Below is the code of how I used callbacks with model.fit_generator:

earlystop = EarlyStopping(monitor='val_acc', patience=50, verbose=1, mode='auto')
check = ModelCheckpoint(model_filename,
                        monitor='val_acc',
                        verbose=1,
                        save_best_only=True,
                        mode='auto')
history = History()
tensorboard = TensorBoard(log_dir='./logs', write_graph=True, write_images=True)

# Fit the model on the batches generated by datagen.flow().
model.fit_generator(train_generator,
                    steps_per_epoch=(nb_train_samples/batch_size),
                    epochs=epochs,
                    validation_data=validation_generator,
                    validation_steps=(nb_validation_samples/batch_size),
                    callbacks=[earlystop,check,history,tensorboard],
verbose=1)

**Comment by netrapathak** [comment]
Below is the code for how I used callbacks with model.fit_generator:

--> Define your callbacks:
earlystop = EarlyStopping(monitor='val_acc', patience=50, verbose=1, mode='auto')
check = ModelCheckpoint(model_filename,
                        monitor='val_acc',
                        verbose=1,
                        save_best_only=True,
                        mode='auto')
history = History()
tensorboard = TensorBoard(log_dir='./logs', write_graph=True, write_images=True)

-->Fit the model on the batches generated by datagen.flow():
model.fit_generator(train_generator,
                    steps_per_epoch=(nb_train_samples/batch_size),
                    epochs=epochs,
                    validation_data=validation_generator,
                    validation_steps=(nb_validation_samples/batch_size),
                    callbacks=[earlystop,check,history,tensorboard],
verbose=1)
